lib/devicetrust/native/device_windows.go

```diff
@@ -494,12 +508,30 @@ func solveTPMEnrollChallenge(
 	if encryptedCredential == nil {
 		return nil, trace.BadParameter("missing encrypted credential in challenge from server")
 	}
-	activationSolution, err := ak.ActivateCredential(
-		tpm,
-		*encryptedCredential,
-	)
-	if err != nil {
-		return nil, trace.Wrap(err, "activating credential with challenge")
+
+	var activationSolution []byte
+	if windows.GetCurrentProcessToken().IsElevated() {
+		log.Debug("TPM: Detected current process is elevated. Will run credential activation in current process.")
+		// If we are running with elevated privileges, we can just complete the
+		// credential activation here.
+		activationSolution, err = ak.ActivateCredential(
+			tpm,
+			*encryptedCredential,
+		)
+		if err != nil {
+			return nil, trace.Wrap(err, "activating credential with challenge")
+		}
+	} else {
+		_, _ = fmt.Fprintln(os.Stderr, "Detected that tsh is not running with elevated privileges. Triggering a UAC prompt to complete the enrollment in an elevated process.")
```

Comment:
I _think_ the linters won't complain on Print-family functions. Ignore-me otherwise.

```suggestion
		fmt.Fprintln(os.Stderr, "Detected that tsh is not running with elevated privileges. Triggering a UAC prompt to complete the enrollment in an elevated process.")
```